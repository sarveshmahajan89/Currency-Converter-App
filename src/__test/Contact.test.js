import React from 'react';

import Contact from '../component/Contact.js'
import {shallow} from "enzyme/build/index";

describe('Contact', function() {
    const component = shallow(<Contact/>);
    it('Contact renders correctly', () => {

        expect(<Contact/>).toMatchSnapshot();
    });

    it('should render Contact correctly with shallow', function () {

        expect(component).toMatchSnapshot();
    });

    it('Contact should render h1 elements correctly', function () {

        const text = component.find('h1').text();
        expect(text).toEqual('Contact us');
    });

    it('Contact should render p elements correctly', function () {

        const text = component.find('p').text();
        expect(text).toEqual('Our team looks forward to answering your questions. Simply fill out the form and we will get back to you as soon as possible.');
    });
});