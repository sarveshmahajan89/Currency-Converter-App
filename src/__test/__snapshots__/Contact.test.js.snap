// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Contact renders correctly 1`] = `<Contact />`;

exports[`Contact should render Contact correctly with shallow 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
      Contact us
    </h1>
    <p>
      Our team looks forward to answering your questions. Simply fill out the form and we will get back to you as soon as possible.
    </p>
  </div>,
  "nodes": Array [
    <div>
      <h1>
        Contact us
      </h1>
      <p>
        Our team looks forward to answering your questions. Simply fill out the form and we will get back to you as soon as possible.
      </p>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useDeferredValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useResponder": [Function],
      "useState": [Function],
      "useTransition": [Function],
    },
    "_element": <Contact />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": Contact {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <div>
      <h1>
        Contact us
      </h1>
      <p>
        Our team looks forward to answering your questions. Simply fill out the form and we will get back to you as soon as possible.
      </p>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Contact />,
}
`;
